File: src/www/__test__/testPost/reader/PostContent.ts
import { PostLine } from "./PostLine";
import { PostSection } from "./PostSection";

export class PostContent {
  private sectionsList: PostSection[] = [];
  private sectionsByName: { [name: string]: PostSection } = {};

  constructor(private postId: string) {}

  getPostId(): string {
    return this.postId;
  }

  addLine(sectionName: string | null, subsectionName: string, line: PostLine) {
    if (sectionName == null) return;

    if (!this.sectionsByName[sectionName]) {
      const newSection = new PostSection(this, sectionName);
      this.sectionsByName[sectionName] = newSection;
      this.sectionsList.push(newSection);
    }

    var section = this.sectionsByName[sectionName];
    section.addLine(subsectionName, line);
  }

  getSections(): PostSection[] {
    return this.sectionsList;
  }

  getPrettyName(): string {
    return `Post: ${this.postId}`;
  }
}

File: src/test/java/com/drpicox/game/testPost/reader/PostContent.java
package com.drpicox.game.testPost.reader;

import java.util.*;

public class PostContent {
    private final Map<String, PostSection> sections = new LinkedHashMap<>();
    private final String postId;

    public PostContent(String postId) {
        this.postId = postId;
    }

    public String getPostId() {
        return postId;
    }

    public void addLine(String sectionName, String subsectionName, PostLine line) {
        if (sectionName == null) return;

        if (!sections.containsKey(sectionName)) {
            sections.put(sectionName, new PostSection(this, sectionName));
        }

        var section = sections.get(sectionName);
        section.addLine(subsectionName, line);
    }

    public Collection<PostSection> getSections() {
        return sections.values();
    }

    public String getPrettyName() {
        return "Post:" + postId;
    }
}
